src/azure-cli-core/azure/cli/core/adal_authentication.py

```diff
@@ -37,11 +37,14 @@ def signed_session(self, session=None):  # pylint: disable=arguments-differ
 
             err = (getattr(err, 'error_response', None) or {}).get('error_description') or ''
             if 'AADSTS70008' in err:  # all errors starting with 70008 should be creds expiration related
-                raise CLIError("Credentials have expired due to inactivity.{}".format(
-                    " Please run 'az login'" if not in_cloud_console() else ''))
+                raise CLIError("Credentials have expired due to inactivity. {}".format(
+                    "Please run 'az login'" if not in_cloud_console() else ''))
             if 'AADSTS50079' in err:
                 raise CLIError("Configuration of your account was changed. {}".format(
-                    " Please run 'az login'" if not in_cloud_console() else ''))
+                    "Please run 'az login'" if not in_cloud_console() else ''))
+            if 'AADSTS50173' in err:
+                raise CLIError("The provided grant has expired due to it being revoked. You might have changed or reset the password. {}".format(
```

Comment:
On a second thought, the term of `provider grant....` is a bit hard to grok for regular users. I re-suggest `The credential data used by CLI has been expired because you might have changed or reset the password. Please clear browser's...`. Thoughts?

Reply:
Sure. This sounds much clearer to me.